### DISCUSSION TITLE
Auto tuning for pseudo absence (random created within model)

### DISCUSSION THREAD

**Discussion by JingpengCao** [biomodhub/biomod2#222] [Q&A] [answered]
Hello guys,

thank you for creating this model. It has been wonderful.

I have create random pseudo absence and want to automatically tune the modeling as the example has mentioned. But it failed. I have also tried to do this with the example dataset. It seems like it could only run with presence-absence BIOMOD data, not with presence-pseudo absence BIOMOD data set.

Is there anyway to do the auto tune for PA dataset?

**Comment by MayaGueguen** [comment, marked as answer]
Hello there,

Yes, indeed, [BIOMOD_Tuning](https://biomodhub.github.io/biomod2/reference/BIOMOD_Tuning.html) function is not fully functional...
You are right, currently, it does not care at all about pseudo-absences if there is some :grimacing: 

The good news is : 
- we are about to make a new release of `biomod2` (4.2-3) that will contain improvements of selection of pseudo-absence and cross-validation datasets
- once it is done, our next step in our development plan is to work both on : 
    -  better integration of single models parameters (improving [BIOMOD_ModelingOptions](https://biomodhub.github.io/biomod2/reference/BIOMOD_ModelingOptions.html))
    - and correction / improvement of [BIOMOD_Tuning](https://biomodhub.github.io/biomod2/reference/BIOMOD_Tuning.html) function with this new way of dealing with paramterization

And we will make sure at that time that it deals correctly (or at least in a transparent way :smile:) with pseudo-absence and cross-validation datasets.

Maya
